File: 프로그래머스/3/43162. 네트워크/네트워크.cpp
#include <string>
#include <vector>

using namespace std;

int nodev[200];

void dfs(int x, vector<vector<int> > &computers) {
    vector<int> comp = computers[x];
    nodev[x] = 1;
    for(int i = 0; i < comp.size(); i++) {
        if(comp[i] == 1 && nodev[i] == 0) {
            dfs(i, computers);
        }
    }
}

int solution(int n, vector<vector<int>> computers) {
    int answer = 0;
    for(int i = 0; i < n; i++) {
        if(nodev[i]) continue;
        dfs(i, computers);
        answer++;
    }
    return answer;
}

File: 프로그래머스/3/43162. 네트워크/네트워크.java
class Solution {
    int[] visited = new int[200];
    void dfs(int[][] computers, int n) {
        int[] comp = computers[n];
        visited[n] = 1;
        for(int i = 0; i < comp.length; i++) {
            if(comp[i] == 1 && visited[i] == 0) {
                dfs(computers, i);
            }
        }
    }
    
    public int solution(int n, int[][] computers) {
        int answer = 0;
        for(int i = 0; i < n; i++) {
            if(visited[i] == 1) continue;
            dfs(computers, i);
            answer++;
        }
        return answer;
    }
}